meson.build

```diff
@@ -0,0 +1,435 @@
+      extra_dep += cc.find_library(x, dirs : vc_lib_dir)
+    endforeach
+    # override libgles implementation
+    libgles = cc.find_library('brcmGLESv2', dirs : vc_lib_dir)
+  endif
+
+  extra_dep += libgles
+endif
+
+# Assembler step
+if src_nasm.length() > 0
+  # Compile asm_defines.c to extract compiler/os-specific offsets
+  # Use the same compilation flags as the other files of the project
+  # except for LTO which must be disabled for effective creation of required symbols
+  asm_defines_lib = static_library('asm_defines',
+    'src/asm_defines/asm_defines.c',
+    include_directories : inc,
+    c_args : cflags,
+    override_options : 'b_lto=false')
+
+  # Generate headers using an awk parser directly on the asm_defines_lib binary
+  awk = find_program('awk', join_paths(m64pwindeppath, 'gawk-3.1.6-1/bin/gawk.exe'))
+  asm_defines_headers = custom_target('asm_defines_headers',
+      input : ['tools/gen_asm_defines.awk', asm_defines_lib],
+      output : ['asm_defines_nasm.h', 'asm_defines_gas.h'],
+      command : [awk, '-v', 'dest_dir=@OUTDIR@', '-f', '@INPUT0@', '@INPUT1@'])
+
+  asflags = []
+
+  # forward PIC mode to assembler files
+  if not get_option('b_staticpic')
+    asflags += ['-d', 'PIC']
+  endif
+
+  # Select appropriate format
+  if 'linux bsd openbsd'.contains(host_system)
+    nasm_fmt = 'elf' + (host_cpu.endswith('64') ? '64' : '')
+  elif host_system == 'darwin'
+    nasm_fmt = 'macho' + (host_cpu.endswidth('64') ? '64' : '')
+  elif host_system == 'windows'
+    nasm_fmt = 'win' + (host_cpu.endswith('64') ? '64' : '32')
+  else
+    error('Unsupported system')
+  endif
+
+  # Forward leading underscore requirement to assmbler files
+  if cc.symbols_have_underscore_prefix()
+    asflags += ['-d', 'LEADING_UNDERSCORE']
+  endif
+
+  # Some more per-host tweaks
+  if host_system == 'windows'
+    if host_cpu == 'x86_64'
+      asflags += ['-d', 'WIN64']
+    endif
+  endif
+
+  # Add include directory where asm_defines headers are generated
+  asflags += ['-I', '@BUILD_DIR@']
+
+  # Assemble each assembly file
+  # BUG?: Can't make it work with generator because depends : asm_defines_headers is not honored
+  # So we use a custom_target
+  nasm = find_program('nasm', join_paths(m64pwindeppath, 'nasm-2.11.06/nasm.exe'))
+  foreach x : src_nasm
+    src += custom_target('@BASENAME@.o',
+        output : ['@BASENAME@.o'],
+        input : x,
+        depfile : '@BASENAME@.dep',
+        depends : asm_defines_headers,
+        command : [nasm, asflags, '-f', nasm_fmt, '-MD', '@DEPFILE@', '-o', '@OUTPUT@', '@INPUT@'])
+  endforeach
+
+#  nasm_gen = generator(nasm,
+#      output : '@BASENAME@.obj',
+#      depfile : '@BASENAME@.dep',
+#      arguments : ['@EXTRA_ARGS@', '-MD', '@DEPFILE@', '-o', '@OUTPUT@', '@INPUT@'])
+#  src += nasm_gen.process(src_nasm, extra_args : asflags, depends : asm_defines_headers)
+endif
+
+# XXX: linking flags to export api symbols
+library('mupen64plus-core',
```

Comment:
The library is called “libmupen64plus”, not “libmupen64plus-core”. Also, this should be shared_library(), not library() (see next comment).

Reply:
Renamed to mupen64plus. However I kept library, as you can change this using the 'default_library' option.